lib/mapper.js

```diff
@@ -112,14 +113,30 @@ function common(facade){
   if (facade.city()) ret.city = facade.city();
   if (facade.region()) ret.region = facade.region();
 
-  if (os && os.toLowerCase() == 'ios') {
+  function mobileDeviceType(){
+    if (!library && !os) return 'unknown';
+    if (contains('ios')) return 'ios';
+    if (contains('android')) return 'android';
+    return 'unknown';
+
+    function contains(str) {
+      return getField(library).toLowerCase().indexOf(str) !== -1
+          || getField(os).toLowerCase().indexOf(str) !== -1;
+    }
+
+    function getField(field) {
+      return typeof field === 'string' ? field : field.name;
+    }
+ }
+
+  if (mobileDeviceType() === 'ios') {
       ret.idfa = facade.proxy('device.idfa');
-  } else if (os && os.toLowerCase() == 'android') {
-      //Older segment clients sent idfa, newer ones send advertisingId
-      ret.adid = facade.proxy('device.idfa');
-      if(!ret.adid) {
-        ret.adid = facade.proxy('device.advertisingId');
-      }
+  }
+
+  if (mobileDeviceType() === 'android') {
+    //Older segment clients sent idfa, newer ones send advertisingId
+    ret.adid = facade.proxy('device.idfa');
+    if (!ret.adid) ret.adid = facade.proxy('device.advertisingId');
```

Comment:
Also potentially something we could move to api-transforms so we don't have to duplicate it in each integration